### PR TITLE
tiflash: Add "marker" for `security.redact_info_log` in tiflash.toml

### PR DESCRIPTION
**Pull request by JaySon-Huang**
### First-time contributors' checklist 

- [x] I've signed [**Contributor License Agreement**](https://cla-assistant.io/pingcap/docs-cn) that's required for repo owners to accept my contribution.

### What is changed, added or deleted? (Required)

https://github.com/pingcap/tiflash/issues/8977

Enhance log densensitation in tiflash.
A similar feature in TiDB introduced since v8.0: https://github.com/pingcap/docs-cn/pull/16769

### Which TiDB version(s) do your changes apply to? (Required)

**Tips for choosing the affected version(s):**

By default, **CHOOSE MASTER ONLY** so your changes will be applied to the next TiDB major or minor releases. If your PR involves a product feature behavior change or a compatibility change, **CHOOSE THE AFFECTED RELEASE BRANCH(ES) AND MASTER**.

For details, see [tips for choosing the affected versions (in Chinese)](https://github.com/pingcap/docs-cn/blob/master/CONTRIBUTING.md#版本选择指南).

- [x] master (the latest development version)
- [x] v8.2 (TiDB 8.2 versions)
- [ ] v8.1 (TiDB 8.1 versions)
- [ ] v8.0 (TiDB 8.0 versions)
- [ ] v7.6 (TiDB 7.6 versions)
- [ ] v7.5 (TiDB 7.5 versions)
- [ ] v7.1 (TiDB 7.1 versions)
- [ ] v6.5 (TiDB 6.5 versions)
- [ ] v6.1 (TiDB 6.1 versions)
- [ ] v5.4 (TiDB 5.4 versions)
- [ ] v5.3 (TiDB 5.3 versions)
- [ ] v5.2 (TiDB 5.2 versions)
- [ ] v5.1 (TiDB 5.1 versions)

### What is the related PR or file link(s)?

- This PR is translated from:
- Other reference link(s): https://github.com/pingcap/tiflash/issues/8977

### Do your changes match any of the following descriptions?

- [ ] Delete files
- [ ] Change aliases
- [ ] Need modification after applied to another branch 
- [ ] Might cause conflicts after applied to another branch

### CODE REVIEW

File: tiflash/tiflash-configuration.md [hunk 1]
```diff
@@ -242,9 +242,11 @@ delta_index_cache_size = 0
 
 ## 安全相关配置，从 v4.0.5 开始生效
 [security]
-    ## 从 v5.0 引入，控制是否开启日志脱敏
-    ## 若开启该选项，日志中的用户数据会以 `?` 代替显示
-    ## 注意，tiflash-learner 对应的安全配置选项为 `security.redact-info-log`，需要在 tiflash-learner.toml 中另外开启
+    ## 从 v5.0 引入，控制是否开启日志脱敏。从 v8.2 开始，支持设置为 false/true/marker。
```
Comment (Oreoxmt) [inline tiflash/tiflash-configuration.md:245]:
```suggestion
    ## 从 v5.0 引入，控制是否开启日志脱敏。可选值为 false、true、marker。其中，marker 从 v8.2.0 开始支持。
```

File: tiflash/tiflash-configuration.md [hunk 2]
```diff
@@ -272,9 +274,9 @@ delta_index_cache_size = 0
     store-batch-retry-recv-timeout = "4ms"
 
 [security]
-    ## 从 v5.0 引入，控制是否开启日志脱敏
-    ## 若开启该选项，日志中的用户数据会以 `?` 代替显示
-    ## 默认值为 false
+    ## 从 v5.0 引入，控制是否开启日志脱敏。支持设置为 false/true。
```
Comment (Oreoxmt) [inline tiflash/tiflash-configuration.md:277]:
```suggestion
    ## 从 v5.0 引入，控制是否开启日志脱敏。可选值为 false、true。
```

**Review by Oreoxmt — APPROVED** [review]
Rest LGTM

**Comment by Oreoxmt** [comment]
@JaySon-Huang Could you please invite a tech reviewer? Thanks!

File: tiflash/tiflash-configuration.md [hunk 3]
```diff
@@ -242,9 +242,11 @@ delta_index_cache_size = 0
 
 ## 安全相关配置，从 v4.0.5 开始生效
 [security]
-    ## 从 v5.0 引入，控制是否开启日志脱敏
-    ## 若开启该选项，日志中的用户数据会以 `?` 代替显示
-    ## 注意，tiflash-learner 对应的安全配置选项为 `security.redact-info-log`，需要在 tiflash-learner.toml 中另外开启
+    ## 从 v5.0 引入，控制是否开启日志脱敏。可选值为 true、false、marker。其中，marker 从 v8.2.0 开始支持。
+    ## 默认值为 false，即对用户日志不做处理。
+    ## 若设置为 true，日志中的用户数据会以 `?` 代替显示。
+    ## 若设置为 marker，日志中的用户数据会被标记符号 `‹ ›` 包裹。输入信息中的 `‹` 会转义成 `‹‹`，`›` 会转义成 `››`。基于标记后的日志，你可以在展示日志时决定是否对被标记信息进行脱敏处理。
```
Comment (qiancai) [inline tiflash/tiflash-configuration.md:248]:
```suggestion
    ## 若设置为 marker，日志中的用户数据会被标记符号 `‹ ›` 包裹。用户数据中的 `‹` 会转义成 `‹‹`，`›` 会转义成 `››`。基于标记后的日志，你可以在展示日志时决定是否对被标记信息进行脱敏处理。
```

Comment (qiancai) [reply tiflash/tiflash-configuration.md:248]: 这里的 "输入信息" 是指用户数据吗

Comment (JaySon-Huang) [reply tiflash/tiflash-configuration.md:248]: 嗯，用户存储的数据或者查询条件中具体的值。“用户数据” 的表达应该清楚些

**Review by qiancai — APPROVED** [review]

**Review by Lloyd-Pottiger — APPROVED** [review]

**Pull request merged**